### PR TITLE
Colored noise in FIRuncFilter and provide utility functions

### PR DESCRIPTION
**Pull request by mgrub**
Requires detailled review of implemented new methods
- check normalization of created ideal autocorrelation 
- check normalization of generated colored noise from white noise
- check execution of propagate-filter (compare sigma as float or np.ndarray)
- review tests

### CODE REVIEW

**Comment by mgrub** [comment]
Next steps: 
- review new implementation with @eichstaedtPTB 
- if `isinstance(sigma_noise, float)`: maintain the probably more performant old implementations

File: PyDynamic/misc/noise.py [hunk 1]
```diff
@@ -1,6 +1,137 @@
...
+* white_gaussian: normal distributed signal amplitudes with equal power spectral density
+* power_law_noise: normal distributed signal amplitudes with power spectrum `~ f^\alpha`
+* power_law_acf: (theoretical) autocorrelation function of power law noise
+* ARMA: autoregressive moving average noise process
+"""
+
 import numpy as np
+from scipy.linalg import toeplitz
+
+__all__ = ['white_gaussian', 'power_law_noise', 'power_law_acf', 'ARMA']
```
Comment (mgrub) [inline PyDynamic/misc/noise.py]: get_alpha is missing in this list

File: PyDynamic/misc/noise.py [hunk 2]
```diff
@@ -1,6 +1,137 @@
...
+
+    # adjust to given mean + std
+    w_filt = mean + std * (w_filt - np.mean(w_filt)) / np.std(w_filt)
+
+    return w_filt
+
+
+def power_law_acf(N, color_value = "white", std = 1, returnMatrix = False):
+    """
+    Return the theoretic autocovariance-matrix (Rww) of different colors of noise. If "alpha" is provided, "color"-argument is ignored.
```
Comment (mgrub) [inline PyDynamic/misc/noise.py]: "alpha" and "color" are now the same argument "color_value"

File: PyDynamic/misc/noise.py [hunk 3]
```diff
@@ -1,6 +1,137 @@
...
+
+
+def power_law_noise(N = None, w = None, color_value = "white", mean = 0, std = 1):
+    """
+    Generate colored noise by
+    * generate white gaussian noise
+    * multiplying its Fourier-transform with f^(alpha/2)
+    * inverse Fourier-transform to yield the colored/correlated noise
+    * further adjustments to fit to specified mean/std
+
```
Comment (mgrub) [inline PyDynamic/misc/noise.py]: parameter and return value descriptions are missing

File: PyDynamic/misc/noise.py [hunk 4]
```diff
@@ -1,6 +1,137 @@
...
+    # (real) fourier transform to get spectrum
+    sp   = np.fft.rfft(w)
+
+    # get index of frequencies
+    # note:
+    # * this gives [1., 2., 3., ..., N+1] (in accordance with Zhivomirov2018)
+    # * ==> not ~ f^alpha, but rather ~ (k+1)^alpha (with a zero-based index k)
+    # * this could be a point for further discussion
+    freq = np.fft.rfftfreq(N, d=1/N) + 1
```
Comment (mgrub) [inline PyDynamic/misc/noise.py]: "freq" should be renamed to "k" to match the cited paper

**Comment by BjoernLudwigPTB** [comment]
This would be the changelog as of now:

---

### Fix
* Set internal state of lfilter (23c9998e873d8d432dd51e17aaf91269890d6387)
* Case: diag and no lowpass (586126fb45c4f980a77872b0df8852fa4099ae10)
* Case: corr and lowpass (4826892f930d2e8c32b1fef634cb7dd530d16303)
* Case: diag and lowpass (e7269cc69d57d17b702247516cc6b425c0885f3e)
* Return single sided acf (deae5072b4bb28772df7b03a44e1006ebdaeca35)

### Documentation
* Add validation script for firuncfilter (448ece9f1c4ba6c48e167841c8c5092549df98ca)

**Comment by mgrub** [comment]
@BjoernLudwigPTB : Please add the follwoing features to the changelog:
feat(FIRuncFilter): input signal of FIRuncFilter can have colored/correlated noise
feat(misc.noise): added utility functions in misc.noise for generation of colored noise and the corresponding right-sided autocorrelation

**Comment by BjoernLudwigPTB** [comment]
> @BjoernLudwigPTB : Please add the follwoing features to the changelog:
> feat(FIRuncFilter): input signal of FIRuncFilter can have colored/correlated noise
> feat(misc.noise): added utility functions in misc.noise for generation of colored noise and the corresponding right-sided autocorrelation

Check!

**Pull request merged**